src/synchronizer.ts

```diff
@@ -144,29 +132,18 @@ export async function createSynchronizer(
     for (const snapshotInfo of snapshotInfoByHash.values()) {
       deploymentsProcessorsQueue
         .add(async () => {
-          const stream = getDeployedEntitiesStreamFromSnapshot(
-            { ...components, processedSnapshots },
-            options,
-            snapshotInfo
-          )
-          for await (const entity of stream) {
-            if (isStopped) {
-              logger.debug('Canceling running sync snapshots stream')
-              return
-            }
-            // schedule the deployment in the deployer. the await DOES NOT mean that the entity was deployed entirely
-            // if the deployer is not synchronous. For example, the batchDeployer used in the catalyst just add it in a queue.
-            // Once the entity is truly deployed, it should call the method 'markAsDeployed'
-            await components.deployer.deployEntity(
-              {
-                ...entity,
-                markAsDeployed: async function () {
-                  components.metrics.increment('dcl_entities_deployments_processed_total', { source: 'snapshots' })
-                  await processedSnapshots.entityProcessedFrom(entity.snapshotHash)
-                },
-                snapshotHash: snapshotInfo.snapshotHash
-              },
-              entity.servers
+          const { greatestEndTimestamp, replacedSnapshotHashes, snapshotHash } = snapshotInfo
+          const shouldProcessSnapshot =
+            greatestEndTimestamp > genesisTimestamp &&
+            (await shouldProcessSnapshotAndMarkAsProcessedIfNeeded(components, snapshotHash, replacedSnapshotHashes)) &&
+            !(await components.snapshotStorage.has(snapshotHash))
```

Comment:
Meybe these three && should be moved inside `shouldProcessSnapshotAndMarkAsProcessedIfNeeded`.

Side question: notice that `shouldProcessSnapshotAndMarkAsProcessedIfNeeded` has a side effect, it adds the snapshot to `processedSnapshotStorage`. Is that intended to happen before the actual `deployEntitiesFromSnapshot` from the next line?

Reply:
Yeah this part is a bit confusing, it's not ideal to have a side effect in that function. It's there just to take advantage of 1. hashes already loaded from db and 2. already calculation logic to mark it as processed. That's why i added the last part of the function name "andMarkAsProcessedIfNeeded".

It is not needed to happen before `deployEntitiesFromSnapshot` as this won't be called in this case.

Reply:
Changed in commit [b7b1fed](https://github.com/decentraland/snapshots-fetcher/pull/106/commits/b7b1fed06bbd421d98daac3ddcd78e78cea832bc)